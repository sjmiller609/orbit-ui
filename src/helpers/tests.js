import { BrowserRouter } from 'react-router-dom'
import { shape } from 'prop-types'
import { mount } from 'enzyme'

// Instantiate router context
const router = {
  history: new BrowserRouter().history,
  route: {
    location: {},
    match: {},
  },
}

const createContext = () => ({
  context: { router },
  childContextTypes: { router: shape({}) },
})

// wrap with router context - to mount RR links correctly
export function mountWrap(node) {
  return mount(node, createContext())
}
