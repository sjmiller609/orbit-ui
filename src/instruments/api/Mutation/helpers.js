const has = (str, find) => ~str.toLowerCase().indexOf(find.toLowerCase())

export const formErrors = ({ error, errors }) => {
  if (!error) return

  let err = JSON.stringify(error).toLowerCase()
  return errors.find(e => {
    if (e.key && has(err, e.key)) return e
    else if (e.allKeys && e.allKeys.every(k => has(err, k))) return e
    else if (e.someKeys && e.someKeys.some(k => has(err, k))) return e
  })
}
