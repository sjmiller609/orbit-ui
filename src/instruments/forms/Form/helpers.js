// faltten and stringify nested keys
export const unpack = (obj, ns, hash) => {
  if (!obj) return
  return Object.entries(obj).reduce((acc, [key, value]) => {
    const keypath = ns ? `${ns}.${key}` : key
    if (value && typeof value === 'object') {
      return unpack(value, keypath, acc)
    }
    acc[keypath] = value
    return acc
  }, hash || {})
}

const roll = (keys, obj, value) => {
  const key = keys[0]
  // case 1 - not obj
  if (keys.length === 1) {
    obj[key] = value
    return obj
  }
  // case 2 - doesn't exist
  if (!obj[key]) {
    obj[key] = isNaN(keys[1]) ? {} : []
  }
  // case 3 - exists
  obj[key] = roll(
    keys.slice(1),
    Array.isArray(obj[key]) ? [...obj[key]] : { ...obj[key] },
    value
  )

  return obj
}

// convert back into obj
export const pack = (obj, type = {}) => {
  return Object.entries(obj).reduce((acc, [key, value]) => {
    const keys = key.split('.')

    return roll(keys, acc, value)
  }, type)
}

export const packChild = ({ name, obj }) => {
  const obj2 = {}
  let isArray
  Object.entries(obj).forEach(k => {
    if (name.length < k[0].length && k[0].indexOf(name) === 0) {
      const n = k[0].slice(name.length + 1)
      // must be true for all matching
      isArray = !isNaN(n.slice(0, n.indexOf('.')))
      obj2[n] = k[1]
    }
  })
  if (Object.keys(obj2).length) return pack(obj2, isArray ? [] : {})
  return null
}

export const removeChild = ({ name, obj = {} }) => {
  const obj2 = {}
  Object.entries(obj).forEach(k => {
    if (name.length >= k[0].length || k[0].indexOf(name) !== 0) {
      obj2[k[0]] = k[1]
    }
  })
  return obj2
}
