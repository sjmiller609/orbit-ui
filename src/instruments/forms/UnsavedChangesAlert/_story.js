import React from 'react'
import { storiesOf } from '@storybook/react'
import { UnsavedChangesAlert } from 'instruments'

storiesOf('Instruments|Forms.UnsavedChangesAlert', module).add(
  'Default',
  () => <UnsavedChangesAlert alert={true} />
)
