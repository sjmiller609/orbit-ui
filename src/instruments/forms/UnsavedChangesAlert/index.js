import React from 'react'
import PropTypes from 'prop-types'

import { Prompt } from 'react-router-dom'

class UnsavedChangesAlert extends React.Component {
  mounted = null
  state = {
    alerted: null,
  }

  render() {
    return (
      <Prompt
        when={this.props.alert && !this.state.alerted}
        message={() => {
          this.setState({ alerted: true })
          return 'You have unsaved changes. Are you sure you wish to leave this page?'
        }}
      />
    )
  }
}

UnsavedChangesAlert.propTypes = {
  alert: PropTypes.bool,
}

export default UnsavedChangesAlert
