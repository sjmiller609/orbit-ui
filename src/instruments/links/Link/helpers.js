// check if url is external
export const externalUrl = url => {
  if (typeof url === 'object') return false
  const path = /^https?:\/\/|^\/\//i
  if (path.test(url) && !~url.indexOf(window.location.origin)) return true
  return
}
