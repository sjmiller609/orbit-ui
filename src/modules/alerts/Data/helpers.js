import { formErrors } from 'instruments'

const errors = [
  {
    key: 'permissions',
    name: 'properties.alert_emails',
    error: 'Contact your workspace admin to upgrade your permissions.',
  },
]

export const handleError = error => formErrors({ error, errors })

export const trimError = error => error.split(':')[1].trim()
