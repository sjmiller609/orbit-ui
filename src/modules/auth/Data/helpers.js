import React from 'react'

import { formErrors, Link } from 'instruments'

const errors = [
  {
    key: 'invalid password',
    name: 'password',
    error: 'Incorrect password',
  },
  {
    key: 'not found',
    name: 'email',
    error: "Hmm, we can't find that email",
  },
  {
    key: 'email already in use',
    name: 'email',
    error: 'That email is already taken.',
  },
  {
    key: 'no password credentials found',
    name: 'email',
    error: 'No password found. Did you mean to login with OAuth?',
  },
  {
    key: 'awaiting email confirmation',
    name: 'email',
    error: (
      <Link to="/resend">
        Your email is not yet verified. Please check your email for a
        verification link.<br /> [Click to resend]
      </Link>
    ),
  },
  {
    key: 'not associated with the specified invite',
    name: 'email',
    error: 'The email you entered does not match your invitation.',
  },
  {
    key: 'permissions',
    name: 'permissions',
    error: 'Contact your Workspace Admin to upgrade your permissions.',
  },
  {
    key: 'Email address in use',
    name: 'email',
    error: (
      <Link to="/login">
        A user with that email address already exists. Click here to login.
      </Link>
    ),
  },
]

export const handleError = error => formErrors({ error, errors })
