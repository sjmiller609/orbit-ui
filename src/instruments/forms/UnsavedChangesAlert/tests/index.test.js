import React from 'react'
import { mountWrap } from 'helpers/tests'
import UnsavedChangesAlert from '../index'

const props = {
  alert: true,
}

describe('UnsavedChangesAlert', () => {
  it('renders correctly', () => {
    const wrapper = mountWrap(<UnsavedChangesAlert {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
