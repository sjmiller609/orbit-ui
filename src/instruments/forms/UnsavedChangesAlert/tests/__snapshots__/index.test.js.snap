// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnsavedChangesAlert renders correctly 1`] = `
<UnsavedChangesAlert
  alert={true}
>
  <Prompt
    message={[Function]}
    when={true}
  />
</UnsavedChangesAlert>
`;
